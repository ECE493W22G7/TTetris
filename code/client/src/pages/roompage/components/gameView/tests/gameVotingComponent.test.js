import React from 'react';
import '@testing-library/jest-dom';
import MockedSocket from 'socket.io-mock';
import RoomPage from 'pages/roompage/roompage';
import { Routes, Route, MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import GameVotingComponent from '../gameVotingComponent';

describe("Testing GameVotingComponent", () => {
    let socket;

    beforeEach(() => {
        socket = new MockedSocket();
        window.gameData = {
            players: {
                "abcd":{
                    nickname: "abc",
                    color: "#FFF",
                    init_ofx: 0,
                    currentPiece: null,
                    isImposter: false,
                    hasEmergency: true,
                    isExiled: false,
                },
                "efgh":{
                    nickname: "efg",
                    color: "#FFF",
                    init_ofx: 5,
                    currentPiece: null,
                    isImposter: false,
                    hasEmergency: true,
                    isExiled: false,
                }
            }
        };
    })

    afterEach(() => {
        jest.restoreAllMocks();
    })

    test("Should show up", () => {
        const tree = renderer
            .create(<GameVotingComponent socket={socket.socketClient} />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    })

    test("Should send 'game_input' on vote pressed", async () => {
        const component = renderer
            .create(<GameVotingComponent socket={socket.socketClient} />)

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        const gameInputCallback = jest.fn();
        socket.on("game_input", gameInputCallback);

        let a = await component.root.findAllByType('button');
        a[0].props.onClick();

        expect(gameInputCallback).toBeCalled();
    })
})