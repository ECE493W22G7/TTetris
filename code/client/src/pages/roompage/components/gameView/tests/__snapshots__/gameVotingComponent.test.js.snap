// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GameVotingComponent Should send 'game_input' on vote pressed 1`] = `
<div
  className="vote-box"
>
  <span
    className="vote-title font-italic text-center"
  >
    Guess the Imposter!
  </span>
  <div
    className="player-select-list"
  >
    <button
      className="btn vote-button-color btn-block"
      onClick={[Function]}
      type="button"
    />
    <button
      className="btn vote-button-color btn-block"
      onClick={[Function]}
      type="button"
    />
  </div>
  <div
    className="voteResults"
  >
    <div>
      : [
      ]
    </div>
    <div>
      : [
      ]
    </div>
  </div>
</div>
`;

exports[`Testing GameVotingComponent Should show up 1`] = `
<div
  className="vote-box"
>
  <span
    className="vote-title font-italic text-center"
  >
    Guess the Imposter!
  </span>
  <div
    className="player-select-list"
  >
    <button
      className="btn vote-button-color btn-block"
      onClick={[Function]}
      type="button"
    />
    <button
      className="btn vote-button-color btn-block"
      onClick={[Function]}
      type="button"
    />
  </div>
  <div
    className="voteResults"
  >
    <div>
      : [
      ]
    </div>
    <div>
      : [
      ]
    </div>
  </div>
</div>
`;
